### PR TITLE
Rename backwards compatiblity to backward compatibility

### LINKED ISSUE #12579: Adhere to WordPress core glossary
In a number of places, documentation uses the term "backwards compatibility" instead of "backward compatibility", which is the officially recommended spelling according to https://make.wordpress.org/core/handbook/glossary/ and https://make.wordpress.org/core/handbook/best-practices/spelling/

Situation is probably similar for other words too.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Docs update, renames backwards compatibility to backward compatibility.
Per [WordPress Spelling Guide](https://make.wordpress.org/core/handbook/best-practices/spelling/)

Fixes #12579 

## Types of changes

Documentation update.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -166,9 +166,9 @@ Maintaining dozens of npm packages is difficult—it can be tough to keep track
 
 The developer who proposes a change (pull request) is responsible to choose the correct version increment (`major`, `minor`, or `patch`) according to the following guidelines:
 
-- Major version X (X.y.z | X > 0) should be changed with any backwards-incompatible/"breaking" change. This will usually occur at the final stage of deprecating and removing of a feature.
-- Minor version Y (x.Y.z | x > 0) should be changed when you add functionality or change functionality in a backwards-compatible manner. It must be incremented if any public API functionality is marked as deprecated.
-- Patch version Z (x.y.Z | x > 0) should be incremented when you make backwards-compatible bug fixes.
+- Major version X (X.y.z | X > 0) should be changed with any backward incompatible/"breaking" change. This will usually occur at the final stage of deprecating and removing of a feature.
+- Minor version Y (x.Y.z | x > 0) should be changed when you add functionality or change functionality in a backward  compatible manner. It must be incremented if any public API functionality is marked as deprecated.
```
Comment (chrisvanpatten) [inline CONTRIBUTING.md]:
```suggestion
- Minor version Y (x.Y.z | x > 0) should be changed when you add functionality or change functionality in a backward compatible manner. It must be incremented if any public API functionality is marked as deprecated.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -166,9 +166,9 @@ Maintaining dozens of npm packages is difficult—it can be tough to keep track
...
+- Patch version Z (x.y.Z | x > 0) should be incremented when you make backward  compatible bug fixes.
```
Comment (chrisvanpatten) [inline CONTRIBUTING.md]:
```suggestion
- Patch version Z (x.y.Z | x > 0) should be incremented when you make backward compatible bug fixes.
```

File: docs/designers-developers/developers/README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ You can also filter certain aspects of the editor; this is documented on the [Ed
 
 **Porting PHP meta boxes to blocks and Gutenberg plugins is highly encouraged!**
 
-Discover how [Meta Box](../../../docs/designers-developers/developers/backwards-compatibility/meta-box.md) support works in Gutenberg.
+Discover how [Meta Box](../../../docs/designers-developers/developers/backward-compatibility/meta-box.md) support works in Gutenberg.
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/README.md]: @dd32 Will changing this mean we need a redirect on the handbook environment if we change this?

**Review by chrisvanpatten — APPROVED** [review]
I have a few small changes, but overall this looks good. I'm approving from my perspective but because this touches code (comments, but still), I may request another review before merging to be extra safe.

**Comment by chrisvanpatten** [comment]
@mkaz can you check the merge conflict on this? Sorry 😬

**Comment by mkaz** [comment]
@chrisvanpatten Rebased. I was debating if I should break up the changes in code and the docs to avoid conflicts and make it easier to review. 🤷‍♂️

**Pull request merged**